index.d.ts

```diff
@@ -0,0 +1,249 @@
+declare module 'ember-concurrency' {
+  import Task from 'ember-concurrency/task';
+
+  type TaskDecorator = PropertyDecorator & {
+    drop(): TaskDecorator;
+    keepLatest(): TaskDecorator;
+    restartable(): TaskDecorator;
+    maxConcurrency(num: number): TaskDecorator;
+    enqueue(): TaskDecorator;
+    withTestWaiter(): PropertyDecorator;
+  };
+
+  export function task(
+    generator: Function
```

Comment:
should this be a `GeneratorFunction` (or w/e the TypeScript-equivalent type name is)?

Reply:
depends on if any type inference could be gained from doing so, like how I'm getting inference with decorators